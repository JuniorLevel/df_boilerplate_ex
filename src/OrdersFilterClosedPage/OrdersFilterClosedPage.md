```jsx { "file": "./OrdersFilterClosedPage.jsx" }

```

```jsx noeditor
import Image1 from './ordersFilter-closed1.png';
import Image2 from './ordersFilter-closed2.png';
<div>
	<div>
		<img src={Image1} />
	</div>
	<div>
		<img src={Image2} />
	</div>
</div>;
```
