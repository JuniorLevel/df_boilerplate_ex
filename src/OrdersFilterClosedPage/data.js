export const sidebarButtons = [
	{
		id: 1,
		title: 'Orders',
		onClick: () => {},
	},
	{
		id: 2,
		title: 'Compare',
		onClick: () => {},
	},
	{
		id: 3,
		title: 'Admin',
		onClick: () => {},
	},
];

export const data = [
	{
		id: 1,
		dataSearch: [] as Array<empty>,
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 2,
		dataSearch: [] as Array<empty>,
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 3,
		dataSearch: [] as Array<empty>,
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 4,
		dataSearch: [] as Array<empty>,
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 5,
		dataSearch: [] as Array<empty>,
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 6,
		dataSearch: [] as Array<empty>,
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
];

export const filters = [
	{
		id: 1,
		title: 'Flow_typed-1',
		item: [{ id: 1, status: 'updated', text: 'lorem-1' }],
	},

	{
		id: 2,
		title: 'Flow_typed-2',
		item: [
			{ id: 1, status: 'changed', text: 'project-1' },
			{ id: 2, status: 'changed', text: 'project-2' },
			{ id: 3, status: 'updated', text: 'project-3' },
		],
	},
	{
		id: 3,
		title: 'Flow_typed-3',
		item: [
			{ id: 1, status: 'updated', text: 'issue-1' },
			{ id: 2, status: 'updated', text: 'issue-2' },
		],
	},
];
