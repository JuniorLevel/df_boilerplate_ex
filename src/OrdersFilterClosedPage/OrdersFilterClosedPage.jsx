// @flow

import React from 'react';
import { FiltersSearchListItem, Layout, OrdersFilterClosed, OrdersFilterProvider, FiltersSearchListContainer } from '@/components';
import { data, filters, sidebarButtons } from './data';

export const OrdersFilterClosedPage = (): React.Node => (
	<OrdersFilterProvider>
		<Layout buttons={sidebarButtons} title="Caption" avatar={{ title: 'AV' }}>
			<OrdersFilterClosed open={false} dialog={false} filters={filters} />
			<FiltersSearchListContainer>
				<FiltersSearchListItem key={data[0].id} listData={data[0]} />
				<FiltersSearchListItem key={data[1].id} listData={data[1]} />
				<FiltersSearchListItem key={data[2].id} listData={data[2]} />
				<FiltersSearchListItem key={data[3].id} listData={data[3]} />
				<FiltersSearchListItem key={data[4].id} listData={data[4]} />
				<FiltersSearchListItem key={6} listData={null} />
			</FiltersSearchListContainer>
		</Layout>
	</OrdersFilterProvider>
);
