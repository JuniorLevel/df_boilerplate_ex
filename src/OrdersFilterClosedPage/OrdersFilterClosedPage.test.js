import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { OrdersFilterClosedPage } from './OrdersFilterClosedPage';
import '../../__mocks__/fileMock';

describe('OrdersFilterClosedPage', () => {
	test('should render OrdersFilterClosedPage component without crashing', () => {
		const { container } = render(<OrdersFilterClosedPage />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<BrowserRouter>
					<OrdersFilterClosedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<BrowserRouter>
					<OrdersFilterClosedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<BrowserRouter>
					<OrdersFilterClosedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
