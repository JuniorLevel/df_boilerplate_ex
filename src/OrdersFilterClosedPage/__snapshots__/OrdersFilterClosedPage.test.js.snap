// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersFilterClosedPage matches the snapshot desktop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      >
        <div
          class="FiltersSearchListContainer__StyledFiltersSearchContainer-sc-65iseu-0 chQval"
        />
      </main>
    </div>
  </div>
</div>
`;

exports[`OrdersFilterClosedPage matches the snapshot laptop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      >
        <div
          class="FiltersSearchListContainer__StyledFiltersSearchContainer-sc-65iseu-0 chQval"
        />
      </main>
    </div>
  </div>
</div>
`;

exports[`OrdersFilterClosedPage matches the snapshot mobile 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      >
        <div
          class="FiltersSearchListContainer__StyledFiltersSearchContainer-sc-65iseu-0 chQval"
        />
      </main>
    </div>
  </div>
</div>
`;
